docs/spec/concepts.rst

```diff
@@ -3,97 +3,216 @@
-          return x * 2
-
-Since the subclasses of ``Enum`` cannot be further subclassed,
-the type of variable ``x`` can be statically inferred in all branches
-of the above example. The same approach is applicable if more than one
-singleton object is needed: one can use an enumeration that has more than
-one value::
-
-  class Reason(Enum):
-      timeout = 1
-      error = 2
-
-  def process(response: str | Reason = '') -> str:
-      if response is Reason.timeout:
-          return 'TIMEOUT'
-      elif response is Reason.error:
-          return 'ERROR'
-      else:
-          # response can be only str, all other possible values exhausted
-          return 'PROCESSED: ' + response
+Static, dynamic, and gradual typing
+-----------------------------------
+
+A **statically-typed** programming language runs a type-checker before running
+a program. The program is required to be well-typed according to the language's
+type system. The type system assigns a type to all expressions in the language
+and verifies that their uses obey the typing rules. Normally, a program that is
+not well-typed (i.e., one that contains a type error) will not run. Java and
+C++ are examples of statically-typed object-oriented languages.
+
+A **dynamically-typed** programming language does not run a type-checker before
+running a program. Instead, it will check the types of values before performing
+operations on them at runtime. This is not to say that the language is
+"untyped". Values at runtime have a type and their uses obey typing rules. Not
+every operation will be checked, but certain primitive operations in the
+language such as attribute access or arithmetic will be. Python was
+historically a dynamically-typed language.
+
+**Gradual typing** is the name for a specific way to combine static and dynamic
+typing. Type-annotated Python allows mixing static and dynamic checking at a
+fine level of granularity. Individual variables, parameters, and returns can
+optionally be given a static type. Data structures such as objects can have a
+mix of static and dynamic-only checking. For example, a dictionary could be
+annotated to have static checking of the key type but only dynamic checking of
+the value type.
+
+In gradual typing, a dynamically typed term is indicated by a special "unknown"
+or "dynamic" type.  In Python, the unknown type is spelled :ref:`Any`. It
+indicates to the static type checker that this term should not be subject to
+static checking.  The system should not signal a static type error for use of a
+term with type :ref:`Any`.  Instead, the term's type will be dynamically
+checked, according to the Python runtime's usual checks on the types of runtime
+values.
+
+This specification describes a gradual type system for Python.
+
+Static, dynamic, and gradual types
+----------------------------------
+
+We will refer to types that do not contain :ref:`Any` as a sub-part as **static
+types**.
+
+:ref:`Any` itself is the **dynamic type**.
```

Comment:
I don't think the fact that all types are checked at runtime in Python is a reason to avoid the term "dynamic type" for Any. Any expresses that we are _relying_ on that "dynamic typing" behavior here, rather than relying on static type checking.

I don't think the use of the term "dynamic" for Any is related to the insertion of runtime checked casts in sound gradually typed languages. The naming is more fundamental to the mix of static and dynamic typing which characterizes gradual type systems, whether sound (inserting runtime checked casts) or not.